bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/discovery/BusGatewayUpnpDiscovery.java

```diff
@@ -0,0 +1,229 @@
+            logger.info("                              |- {} ({})", devInfo.modelName, devInfo.manufacturer);
+        }
+        ThingUID thingId = generateThingUID(devInfo);
+        if (thingId != null) {
+            String host = devInfo.host;
+            if (host != null) {
+                Map<String, Object> properties = new HashMap<>(4);
+                String label = "BUS Gateway (" + thingId.getId().split("-")[0] + ")";
+                try {
+                    label = ((!("".equals(devInfo.friendlyName))) ? devInfo.friendlyName : "NO_NAME");
+                    label = label + " (" + devInfo.modelName + ", " + devInfo.modelNumber + ", " + devInfo.host + ")";
+                } catch (Exception e) {
+                    logger.warn("==OWN:UPnP== Exception while getting devInfo for device UDN={}. Exception={}",
+                            devInfo.udn, e.getMessage());
+                }
+                properties.put(OpenWebNetBindingConstants.CONFIG_PROPERTY_HOST, host);
+                properties.put(OpenWebNetBindingConstants.PROPERTY_FIRMWARE_VERSION, devInfo.modelNumber);
+                properties.put(OpenWebNetBindingConstants.PROPERTY_MODEL, devInfo.modelName);
+                properties.put(OpenWebNetBindingConstants.PROPERTY_SERIAL_NO, devInfo.serialNumber);
+                DiscoveryResult result = DiscoveryResultBuilder.create(thingId).withProperties(properties)
+                        .withRepresentationProperty(OpenWebNetBindingConstants.PROPERTY_SERIAL_NO).withLabel(label)
+                        .build();
+                UDN udn = devInfo.udn;
+                String udnStr;
+                if (udn != null) {
+                    udnStr = udn.getIdentifierString();
+                } else {
+                    udnStr = null;
+                }
+                logger.info("Created a DiscoveryResult for gateway '{}' (UDN={})", devInfo.friendlyName, udnStr);
+                return result;
+            } else {
+                logger.warn("Could not get host for device (UDN={})", devInfo.udn);
+                return null;
+            }
+        } else {
+            return null;
+        }
+    }
+
+    @Override
+    public @Nullable ThingUID getThingUID(RemoteDevice device) {
+        return generateThingUID(new DeviceInfo(device));
+    }
+
+    /**
+     * Returns a ThingUID for supported devices from already extracted DeviceInfo
+     *
+     * @param devInfo the device info
+     * @return a new ThingUID, or null if the device is not supported by the binding
+     */
+    private @Nullable ThingUID generateThingUID(DeviceInfo devInfo) {
+        if (devInfo.isBTicino) {
+            UDN udn = devInfo.udn;
+            String idString = null;
+            if (udn != null) {
+                idString = udn.getIdentifierString();
+                if (idString != null) {
+                    BusGatewayId gwId = BusGatewayId.fromValue(idString.split("-")[1]);
```

Comment:
Are you sure there is always a dash in the ID? Better add a length check to the array. Same for below.